src/Core/src/Handlers/Application/ApplicationHandler.iOS.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Runtime.InteropServices;
+using Foundation;
+using Microsoft.Extensions.Logging;
+using UIKit;
+
+namespace Microsoft.Maui.Handlers
+{
+	public partial class ApplicationHandler : ElementHandler<IApplication, UIApplicationDelegate>
+	{
+		public static void MapTerminate(ApplicationHandler handler, IApplication application, object? args)
+		{
+#if __MACCATALYST__
+			NSApplication.SharedApplication.Terminate();
```

Comment:
Mac Catalyst is basically a macOS app that uses a set of "cusomt controls" that just happened to match the API of UIKit. So, a `NSApplication` exists as well as there are ways to get the `NSWindow` from `UIWindow`.